Python/ast.c

```diff
@@ -2815,27 +2827,16 @@ ast_for_call(struct compiling *c, const node *n, expr_ty func, bool allowgen)
                         break;
                     expr_node = CHILD(expr_node, 0);
                 }
-                if (TYPE(expr_node) == lambdef) {
-                    // f(lambda x: x[0] = 3) ends up getting parsed with LHS
-                    // test = lambda x: x[0], and RHS test = 3.  Issue #132313
-                    // points out that complaining about a keyword then is very
-                    // confusing.
+                if (TYPE(expr_node) != NAME || is_forbidden_name(STR(expr_node))) {
                     ast_error(c, chch,
-                            "lambda cannot contain assignment");
-                    return NULL;
-                }
-                else if (TYPE(expr_node) != NAME) {
-                    ast_error(c, chch,
-                              "keyword can't be an expression");
+                              "expression cannot contain assignment, "
+                              "perhaps you meant \"==\"?");
```

Comment:
As a matter of style, I don't think we should use the second person ("you") in error messages.

Reply:
I agree. Could you please suggest better error message?

I based this error message on the error messages for Python 2 style `print`. There are also other error messages that contain "you":

* `"Missing parentheses in call to 'print'. Did you mean print(%U%s)?"`
* `"unsupported operand type(s) for %.100s: '%.100s' and '%.100s'. Did you mean \"print(<message>, file=<output_stream>)\"?"`
* `"if you give only one argument to maketrans it must be a dict"`
* `"new-line character seen in unquoted field - do you need to open the file in universal-newline mode?"`
* `"iterator should return strings, not %.200s (did you open the file in text mode?)"`
* `"utime: you may specify either 'times' or 'ns' but not both"`
* `"Binding %d has no name, but you supplied a dictionary (which has only names)."`

Perhaps we should make them less personalized too. But this is other issue.